[SPARK-39853][CORE] Support stage level task resource profile for standalone cluster when dynamic allocation disabled (#37268)
apache/spark: docs/spark-standalone.md

# Stage Level Scheduling Overview

Stage level scheduling is supported on Standalone:

> - When dynamic allocation is disabled: It allows users to specify different task resource requirements at stage level, and tasks with different task resource requirements will share executors with `DEFAULT_RESOURCE_PROFILE`.

Reviewer: ```suggestion
- When dynamic allocation is disabled: It allows users to specify different task resource requirements at the stage level and will use the same executors requested at startup.
```

Author: Thanks for the suggestions.
